specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/preview/2021-03-31-preview/CommunicationIdentity.json

```diff
@@ -0,0 +1,118 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationIdentityClient",
+    "description": "Azure Communication Identity Service",
+    "version": "2021-03-31-preview"
+  },
+  "paths": {
+    "/m365identities/:issueAccessToken": {
```

Comment:
Why do we not use the same style of format as the acs identity token (`/identities/{id}/:issueAccessToken`)?

Reply:
Copy from discussion on Teams :
They are different identities - ACS vs. AAD
	The payload is different
	The flow on service side is different
	In future they will have different auth (ACS: HMAC/MI vs AAD: composite of HMAC/MI + AAD).
	You'd need to ask Contoso to construct MRI for the AAD identity to have it consistent with the old API.

Reply:
I was not asking to use the same route, just pointing out that the format is different. They do have a different payload and different purpose. `/identities/{id}/:issueAccessToken` has the id in the route, but the new one does not.

Reply:
Maybe just `/tokens/:exchange` ? The POST body would contain the JWT token and the service can infer if it understands the passed token, validate it and see if it can exchange it.

That way it would also be easier to support more token types besides AAD in the future.

The one counter-argument to a `/tokens` route is that `:issueAccessToken` and `:revokeAccessTokens` are already on the `/identities` route 🤔

Reply:
What about `tokenExchange/:exchangeAadToken`?
1. `tokenExchange` makes it clear that this is not the resource for communication tokens
1. `exchangeAadToken` ensures that we are future proof as not all tokens might give us enough context to identify their source

Reply:
/teams/me/:issueAccessToken ??.